app/javascript/components/filters/FilterSearch.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <div class="height-100">
+    <div class="relative">
+      <input type="text" v-model="searchTerm" @keyup.enter="enterPressed()" class="filter--search">
+      <i class="filter--search-icon" v-show="noSearchTerm" @click="enterPressed()"></i>
+      <i class="filter--search-icon close" v-show="!noSearchTerm" @click="clear()"></i>
+    </div>
+
+    <ul class="ul-unstyled filter__options-list filter__options-list--search">
+      <li v-for="option in options" class="filter--search-option">
+        <p 
+          v-show="matches(option)" 
+          @click="clickOption(option)"
+          class="filter--search-label no-margin"
+          :class="{ 'filter--search-label--active' : isActive(option) }">
+          
+          {{ option }}
+        </p>
+      </li>
+    </ul>
+  </div>
+</template>
+
+<script>
+  import { eventHub } from '../../metadata.js'
+
+  export default {
+    name: 'filter-search',
+
+    props: {
+      options: {
+        required: true,
+        type: Array
+      },
+      name: {
+        required: true,
+        type: String
+      },
+      title: {
+        required: true,
+        type: String
+      },
+      type: {
+        required: true,
+        type: String
+      },
+      eventName: {
+        required: true,
+        type: String
+      }
+    },
+
+    data () {
+      return {
+        searchTerm: '',
+        isSelected: null
+      }
+    },
+
+    computed: {
+      noSearchTerm () {
+        return this.searchTerm.length == 0
+      }
+    },
+
+    methods: {
+      isActive (option) {
+        return option == this.isSelected
+      },
+
+      optionId (option) {
+        return option.replace(' |(|)|_', '-').toLowerCase()
```

Comment:
I've seen this also in the FilterRadio file. Can you please explain why is this needed?
Also, is it possible to create a shared function that works across multiple files so to make the code more DRY?